docs/guides/installation.md

```diff
@@ -226,55 +225,75 @@ flux bootstrap gitlab \
   --path=clusters/my-cluster
 ```
 
-### Generic Git Server
+### Air-gapped Environments
 
-For other Git providers such as Bitbucket, Gogs, Gitea, Azure DevOps, etc you can manually setup the repository and the deploy key.
+To bootstrap Flux on air-gapped environments without access to github.com and ghcr.io, first you'll need 
+to download the `flux` binary, and the container images from a computer with access to internet.
 
-Create a Git repository and clone it locally:
+List all container images:
 
-```sh
-git clone ssh://<host>/<org>/my-repository
-cd my-repository
+```console
+$ flux install --export | grep ghcr.io
+
+image: ghcr.io/fluxcd/helm-controller:v0.8.0
+image: ghcr.io/fluxcd/kustomize-controller:v0.9.0
+image: ghcr.io/fluxcd/notification-controller:v0.9.0
+image: ghcr.io/fluxcd/source-controller:v0.9.0
 ```
 
-Create a directory inside the repository:
+Pull the images locally and push them to your container registry:
 
 ```sh
-mkdir -p ./clusters/my-cluster/flux-system
+docker pull ghcr.io/fluxcd/source-controller:v0.9.0
+docker tag ghcr.io/fluxcd/source-controller:v0.9.0 registry.internal/fluxcd/source-controller:v0.9.0
+docker push registry.internal/fluxcd/source-controller:v0.9.0
 ```
 
-Generate the Flux manifests with:
+Copy `flux` binary to a computer with access to your air-gapped cluster,
+and create the pull secret in the `flux-system` namespace:
 
 ```sh
-flux install --version=latest \
-  --export > ./clusters/my-cluster/flux-system/gotk-components.yaml
+kubectl create ns flux-system
+
+kubectl -n flux-system create secret generic regcred \
+    --from-file=.dockerconfigjson=/.docker/config.json \
+    --type=kubernetes.io/dockerconfigjson
 ```
 
-If your cluster must pull images from a private container registry, first you should pull
-the toolkit images from GitHub Container Registry and push them to your registry, for example:
+Finally, bootstrap Flux using the images from your private registry:
 
 ```sh
-docker pull ghcr.io/fluxcd/source-controller:v0.2.0
-docker tag ghcr.io/fluxcd/source-controller:v0.2.0 registry.internal/fluxcd/source-controller:v0.2.0
-docker push registry.internal/fluxcd/source-controller:v0.2.0
+flux bootstrap <GIT-PROVIDER> \
+  --registry=registry.internal/fluxcd \
+  --image-pull-secret=regcred
```

Comment:
I think this should be:

```suggestion
flux bootstrap <GIT-PROVIDER> \
  --registry=registry.internal/fluxcd \
  --image-pull-secret=regcred \
  --token-auth
```

as without access to github.com we are unable to upsert the deploy key?